### DISCUSSION TITLE
How does Openlib send requests and receive responses from Anna's Archive to parse download links?

### DISCUSSION THREAD

**Discussion by RoryGoddard** [dstark5/Openlib#85] [Q&A]
Hi there, 

I should preface this by saying I am a beginner developer still learning the ropes. I am proficient with Python, HTML, and CSS. I have been trying to develop a script using selenium that would search for a particular book on Anna's Archive and then download it locally, however, I cannot get past the cloudlfare check despite various attempts. 

I am guessing you guys found a different way to do things rather than webscraping? In any case, I would appreciate any and all advice with this and I am more than happy to become a contributor to Openlib if there is anything I could potentially do. 

Thanks in advance!

**Comment by dstark5** [comment]
Hi @RoryGoddard , 
Sorry for the delayed reply.
Just make the user to fill the captcha and use the httpOnly "cf_clearance" cookie with the request to get the mirror links
